lib/biker.rb

```diff
@@ -1,3 +1,27 @@
 class Biker
+  attr_reader :name,
+              :rides,
+              :max_distance,
+              :acceptable_terrain
+
+  def initialize(name, max_distance)
+    @name = name
+    @rides = Hash.new([])
+    @acceptable_terrain = []
+    @max_distance = max_distance
+  end
+
+  def learn_terrain!(terrain)
+    @acceptable_terrain << terrain
+  end
+
+  def log_ride(ride, time)
+    if @acceptable_terrain.include?(ride.terrain) && ride.total_distance <= @max_distance
+      @rides[:ride] << time
```

Comment:
So close! This is where things got weird when using the default [] in the hash set up. When I was playing around with it later, it was adding the values twice or not at all. So, it's better to stick to the basics: set up a hash with {} and then use a conditional to check if the key is present or not.